task/inspect-image/0.1/inspect-image.yaml

```diff
@@ -70,24 +70,51 @@ spec:
       if [[ "${IMAGE_URL}" == *":"*"@"* ]]; then
         IMAGE_URL="${IMAGE_URL/:*@/@}"
       fi
-      echo "Inspecting manifest for source image ${IMAGE_URL}."
-      skopeo inspect --no-tags docker://"${IMAGE_URL}" > $IMAGE_INSPECT 2> stderr.txt || true
-      skopeo inspect --no-tags --raw docker://"${IMAGE_URL}" > $RAW_IMAGE_INSPECT 2>> stderr.txt || true
 
-      if [ ! -z $(cat stderr.txt) ]; then
-        echo "Skopeo inspect failed, inspect-image test encountered the following error:"
-        cat stderr.txt
-        note="Task $(context.task.name) failed: Encountered errors while inspecting image. For details, check Tekton task log."
-        TEST_OUTPUT=$(make_result_json -r ERROR -t "$note")
-        echo "${TEST_OUTPUT}" | tee $(results.TEST_OUTPUT.path)
-        exit 0
+      status=-1
+      max_run=5
+      sleep_sec=10
+      for run in $(seq 1 $max_run); do
+        [ "$run" -gt 1 ] && sleep $sleep_sec  # skip last sleep
+        echo "Inspecting manifest for source image ${IMAGE_URL} (try $run/$max_run)."
+        $(skopeo inspect --no-tags docker://"${IMAGE_URL}" > $IMAGE_INSPECT)
```

Comment:
I read about `$(...)`syntax in Bash which is called command substitution, you now add this. How did substitution work correctly so far without this?

Reply:
There was `|| true` before, so it always was true and never failed.
I used `$()` which runs in subshell so I could collect return error code

Reply:
Thanks for the explanation :+1:

Reply:
you are right, it's wrong. Tekton by default injects `set -e`
I'm going to rework this

Reply:
done